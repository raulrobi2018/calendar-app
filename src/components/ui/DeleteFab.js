import React from "react";
import {useDispatch, useSelector} from "react-redux";
import {eventDelete} from "../../actions/events";

export const DeleteFab = () => {
    const {activeEvent} = useSelector((state) => {
        return state.calendar;
    });

    const dispatch = useDispatch();

    const handleClick = () => {
        dispatch(eventDelete(activeEvent));
    };
    return (
        <button className="btn btn-danger fab btn-del" onClick={handleClick}>
            <i className="fas fa-trash"></i>
        </button>
    );
};
