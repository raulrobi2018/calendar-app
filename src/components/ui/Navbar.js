import React from "react";
import {useDispatch, useSelector} from "react-redux";
import {startLogout} from "../../actions/auth";

export const Navbar = () => {
    const dispatch = useDispatch();
    const {name} = useSelector((state) => state.auth);

    const handleLogout = () => {
        //Al ejecutar el startLogout me saca porque cambia el uid y como
        //el AppRouter est√° pendiente de los cambios del uid, entonces se vuelve a
        //redibujar las rutas entonces el uid es null y lo saca al login
        dispatch(startLogout());
    };

    return (
        <div className="navbar navbar-dark bg-dark mb-4">
            <span className="navbar-brand">{name}</span>

            <button className="btn btn-outline-danger" onClick={handleLogout}>
                <i className="fas fa-sign-out-alt"></i>
                <span> Salir</span>
            </button>
        </div>
    );
};
