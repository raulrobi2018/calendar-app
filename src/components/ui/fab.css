.fab {
    border-radius: 100%;
    bottom: 25px;
    font-size: 20px;
    padding: 15px;
    position: fixed;
    cursor: pointer;
}

.btn-add {
    right: 25px;
}

.btn-del {
    left: 25px;
}