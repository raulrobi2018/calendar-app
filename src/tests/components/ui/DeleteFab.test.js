import React from "react";
import {mount, configure} from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import {Provider} from "react-redux";
import {MemoryRouter} from "react-router-dom";

import {DeleteFab} from "../../../components/ui/DeleteFab";

//Configuración necesaria para que funcione mount con React 17
configure({adapter: new Adapter()});
import configureStore from "redux-mock-store"; //ES6 modules
import thunk from "redux-thunk";

// Configuracion necesaria para probar dispatch
const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {};
const store = mockStore(initState);

//Reemplaza la funcionalidad del dispatch del store por una función jest
store.dispatch = jest.fn();

const wrapper = mount(
    <Provider store={store}>
        <MemoryRouter>
            <DeleteFab />
        </MemoryRouter>
    </Provider>
);

describe("Testing DeleteFab component", () => {
    test("should display correctly", () => {
        expect(wrapper).toMatchSnapshot();
    });
});
