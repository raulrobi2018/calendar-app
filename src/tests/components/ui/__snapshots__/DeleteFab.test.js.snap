// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing DeleteFab component should display correctly 1`] = `
<button
  className="btn btn-danger fab btn-del"
  onClick={[Function]}
>
  <i
    className="fas fa-trash"
  />
</button>
`;
